src/main/java/build/buildfarm/worker/resources/ResourceDecider.java

```diff
@@ -63,6 +63,11 @@ public static ResourceLimits decideResourceLimitations(
       limits.cpu.max = 1;
     }
 
+    if (limitGlobalExecution) {
+      limits.cpu.min = Math.max(limits.cpu.min, 1);
```

Comment:
I think you want Math.min here and below. Also, I recall 0 being specified as the 'all available' option, where individual action limits weren't specified.